task5-import-to-s3/task.md

```diff
@@ -0,0 +1,60 @@
+# Task 5 (AWS Simple Storage Service - S3)
+
+**After Lecture 5 "Amazon Simple Storage Service
+(Amazon S3)"**
+
+## Prerequisites
+---
+
+- The task is a continuation of Homework 4 and should be done in the same repos
+- Install the latest version of AWS SDK (https://aws.amazon.com/ru/sdk-for-node-js/)
+- Install the CSV parser package (https://www.npmjs.com/package/csv-parser)
+- Sign in into AWS Console and create a new S3 bucket with a folder, called uploaded
+- Pull latest changes from nodejs-aws-fe (https://github.com/rolling-scopes-school/nodejs-aws-fe)
+
+## TASK 5.1
+---
+
+Create a new service, called **import-service**, with a its own **serverless.yml** file.
+
+Create a lambda function in that **serverless.yml** file, called **importProductsFile**, which will be triggered by the HTTP **GET** method.
+
+The requested URL should be **/import**.
+
+The file name will be passed in a **query string** as a **name** parameter and should be described in the **serverless.yml** file as a **request parameter**.
+
+Update **serverless.yml** with policies to allow lambda functions to interact with **S3**.
+
+A new **Signed URL** should be created with the following **Key: `uploaded/${fileName}`**.
+
+The response from the lambda should be the created **Signed URL**.
+
+The lambda endpoint should be integrated with the frontend by updating **import** property of the API paths configuration.
+
+## TASK 5.2
+---
+
+Create a lambda function under the same **serverless.yml** file, called **importFileParser**, which will be triggered by the S3 **s3:ObjectCreated:*** event.
+
+Configure the event to be triggered only by changes in the **uploaded** folder in S3.
+
+The lambda function should use the **Readable stream** to get an object from S3, parse it via **csv-parser** and log each record to be shown in **CloudWatch**.
+
+The response should be a correct HTTP response code.
+
+## Evaluation criteria
+---
+
+Reviewers should verify the lambda functions by invoking them through provided URLs.
+ 
+- **1** - File **serverless.yml** contains configuration for **importProductsFile** function
+- **3** - The **importProductsFile** lambda function returns a correct response which can be used to upload a catalog into the **S3** bucket
+- **4** - Frontend application is integrated with **importProductsFile** lambda
+- **5** - The **importFileParser** lambda function is implemented and **serverless.yml** contains configuration for the lambda
+
+## Additional (optional) tasks
+---
+
+- **+1** - **async/await** is used in lambda functions
+- **+1** - **importProductsFile** lambda is covered by **unit** tests (**aws-sdk-mock** can be used to mock S3 methods - https://www.npmjs.com/package/aws-sdk-mock)
+- **+1** - At the end of the **stream** the lambda function should move the catalog from the **uploaded** folder into the **parsed** folder
```

Comment:
What should be as a result, uploaded should be deleted? or file should be deleted? What is partial?

Reply:
Added some explanation for this task